tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -150,6 +154,11 @@ public void CloseSpan(Span span)
                     }
                 }
 
+                if (Security.Instance.Enabled && WafSecurityEvents != null)
```

Comment:
`WafSecurityEvents` will only be non-null if ASM was enabled, right? So can we skip the `Security.Instance.Enabled` check entirely? 🤔 I think it's reasonable to do so and is safe given that all this does is serialize some json?

Reply:
Yes, fixed in 937873794